### PR TITLE
[JENKINS-54151] - Fix compatibility with JCasC 1.0, drop support of pre-release versions

### PR DESCRIPTION
**Pull request by oleg-nenashev**
FYI @jglick 

Once the incrementals is ready, we can use this branch to verify the fix

### CODE REVIEW

File: demo/casc/packager-config.yml [hunk 1]
```diff
@@ -6,27 +6,27 @@ bundle:
...
   artifactId: "jenkins-war"
   source:
-    version: 2.107.3
+    version: 2.121.3
 plugins:
   - groupId: "io.jenkins"
     artifactId: "configuration-as-code"
     source:
-      version: 0.8-alpha-SNAPSHOT
+      version: 1.0-rc3
```
Comment (jglick) [inline demo/casc/packager-config.yml]: pending https://github.com/jenkinsci/configuration-as-code-plugin/pull/509

File: demo/casc/packager-config.yml [hunk 2]
```diff
@@ -35,6 +35,6 @@ systemProperties: {
     jenkins.model.Jenkins.slaveAgentPort: "50000",
     jenkins.model.Jenkins.slaveAgentPortEnforce: "true"}
 casc:
-  - id: "initScripts"
+  - id: "casc"
```
Comment (oleg-nenashev) [inline demo/casc/packager-config.yml]: unrelated, just prevents confusion. It is not a script

File: demo/casc/packager-config.yml (hunk 1 above)
Comment (jglick) [reply demo/casc/packager-config.yml]: Update to 1.0 and this should be ready to merge.

**Comment by oleg-nenashev** [comment]
@jrawlings FYI this patch deprecates support of the pre-release JCasC versions. 
I will create a downstream fix for https://github.com/jenkins-x/jenkins-x-serverless

**Comment by oleg-nenashev** [comment]
Will merge it, may need a new version for the tomorrow's hackfest

**Pull request merged**